### PR TITLE
packaging: Enable more tests in the spec file

### PR DESCRIPTION
**Pull request by LecrisUT**

### CODE REVIEW

**Comment by LecrisUT** [comment]
Interesting, I've enabled `git` tests and they seem to be failing in copr :thinking:

**Comment by henryiii** [comment]
The version of hatch-get-fancy-readme is too old. 22.5 added substitutions, so that's probably when `config.substitutions` was added too. 22.3 is the one being used.

**Comment by henryiii** [comment]
If you want to support the older version too, this line:

https://github.com/scikit-build/scikit-build-core/blob/2a127fee78ef288c15b1f8ac7e0c8a0e7631bc8e/src/scikit_build_core/metadata/fancy_pypi_readme.py#L39

Would change to just a single arg (config.fragments) if using the older plugin. We should also set a minimum in our dependencies based on what we go with.

**Comment by LecrisUT** [comment]
Thanks, I think the best solution for this is to make a conditional patch based on the package version j

File: tests/test_dynamic_metadata.py [hunk 1]
```diff
@@ -143,7 +143,7 @@ def test_plugin_metadata():
     )
 
     assert set(GetRequires().dynamic_metadata()) == {
-        "hatch-fancy-pypi-readme",
+        "hatch-fancy-pypi-readme>=22.3",
```
Comment (henryiii) [inline tests/test_dynamic_metadata.py:146]: Ah, thanks, I thought this was there somewhere, but missed it.

File: src/scikit_build_core/metadata/fancy_pypi_readme.py [hunk 2]
```diff
@@ -33,15 +33,18 @@ def dynamic_metadata(
...
     )
 
+    # Version 22.3 does not have fragment support
     return {
         "readme": {
             "content-type": config.content_type,
-            "text": build_text(config.fragments, config.substitutions),
+            "text": build_text(config.fragments, config.substitutions)
+            if hasattr(config, "substitutions")
+            else build_text(config.fragments),  # type: ignore[call-arg]
```
Comment (henryiii) [inline src/scikit_build_core/metadata/fancy_pypi_readme.py:42]:
```suggestion
            # pylint: disable-next=no-value-for-parameter
            else build_text(config.fragments),  # type: ignore[call-arg]
```

**Comment by LecrisUT** [comment]
/packit test. Those are some weird test failures on packit side. I'll get someone to look into it tomorrow

**Comment by LecrisUT** [comment]
I've checked the errors and it seems the important tests in testing-farm are passing. The linting error seems to be a bug. I've forwarded this upstream

**Review by henryiii — APPROVED** [review]

**Pull request merged**